### DISCUSSION TITLE
Unable to compile example using `cargo 3ds build`

### DISCUSSION THREAD

**Discussion by budde25** [rust3ds/ctru-rs#130] [Q&A] [answered]
My system is a M1 Macbookpro running 13.4.1 (22F82)

I have followed the getting started guide installing the required libs, Making sure both DEVKITPRO and DEVKITARM are in my PATH. 

however when I try running a build command: 
`cargo 3ds build --example hello-both-screens`

I get:
```
error: linking with `arm-none-eabi-gcc` failed: exit status: 1

lots of environment printing, then  ...
  = note: arm-none-eabi-gcc: fatal error: cannot read spec file '3dsx.specs': No such file or directory
          compilation terminated. 
```

I am able to find the file manually in my devkitpro install here:
`/opt/devkitpro/devkitARM/arm-none-eabi/lib/3dsx.specs`

Any info is appreciated thanks 😃 !

**Comment by Meziu** [comment, marked as answer]
Are you sure that’s the only “arm-none-eabi-gcc” in your system? The one in dkP should automatically include the file.

I’m also on Mac (Intel) so I don’t know if the ARM platform includes that compiler (or if you downloaded a different version anywhere).

Do:
```
where arm-none-eabi-gcc
```

**Reply by budde25** [reply]
Ahh great catch it looks like I do have one though homebrew:
`/opt/homebrew/bin/arm-none-eabi-gcc`
Removing that from my path and it all built great! Thanks 😃!

**Reply by Meziu** [reply]
Glad I could help! Sadly we rely on the PATH to be correct (which isn't the best way to handle such cases) because we wanted to avoid writing custom code in rustc to find the `dkP`one. It's an issue known to us which I'd like to solve some day.
